In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

pd.set_option("display.max_columns", None)
pd.set_option("display.max_colwidth", 100)

In [2]:
df = pd.read_csv("Dataset.csv")
df.head()

,show_id,type,title,director,country,date_added,release_year,rating,duration,listed_in
0,s1,Movie,Dick Johnson Is Dead,Kirsten Johnson,United States,9/25/2021,2020,PG-13,90 min,Documentaries
1,s3,TV Show,Ganglands,Julien Leclercq,France,9/24/2021,2021,TV-MA,1 Season,"Crime TV Shows, International TV Shows, TV Action & Adventure"
2,s6,TV Show,Midnight Mass,Mike Flanagan,United States,9/24/2021,2021,TV-MA,1 Season,"TV Dramas, TV Horror, TV Mysteries"
3,s14,Movie,Confessions of an Invisible Girl,Bruno Garotti,Brazil,9/22/2021,2021,TV-PG,91 min,"Children & Family Movies, Comedies"
4,s8,Movie,Sankofa,Haile Gerima,United States,9/24/2021,1993,TV-MA,125 min,"Dramas, Independent Movies, International Movies"


In [3]:
print("Daraset Shape:", df.shape)

print("\nColumn Names:")
print(df.columns.tolist())

print("\nData Types:")
print(df.dtypes)

Daraset Shape: (8790, 10)

Column Names:
['show_id', 'type', 'title', 'director', 'country', 'date_added', 'release_year', 'rating', 'duration', 'listed_in']

Data Types:
show_id           str
type              str
title             str
director          str
country           str
date_added        str
release_year    int64
rating            str
duration          str
listed_in         str
dtype: object


In [4]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 8790 entries, 0 to 8789
Data columns (total 10 columns):
 #   Column        Non-Null Count  Dtype
---  ------        --------------  -----
 0   show_id       8790 non-null   str  
 1   type          8790 non-null   str  
 2   title         8790 non-null   str  
 3   director      8790 non-null   str  
 4   country       8790 non-null   str  
 5   date_added    8790 non-null   str  
 6   release_year  8790 non-null   int64
 7   rating        8790 non-null   str  
 8   duration      8790 non-null   str  
 9   listed_in     8790 non-null   str  
dtypes: int64(1), str(9)
memory usage: 686.8 KB


In [5]:
missing_values = df.isnull().sum()
print(missing_values)

show_id         0
type            0
title           0
director        0
country         0
date_added      0
release_year    0
rating          0
duration        0
listed_in       0
dtype: int64


In [6]:
missing_percentage = (df.isnull().sum() / len(df)) * 100

missing_summary = pd.DataFrame({
    "Missing Values": missing_values,
    "Missing Percentage": missing_percentage
})

missing_percentage

show_id         0.0
type            0.0
title           0.0
director        0.0
country         0.0
date_added      0.0
release_year    0.0
rating          0.0
duration        0.0
listed_in       0.0
dtype: float64

In [7]:
duplicate_count = df.duplicated().sum()
print("Number of duplicate rows:", duplicate_count)

Number of duplicate rows: 0


In [8]:
df = df.drop_duplicates().reset_index(drop=True)
print("Shape after removing duplicates:", df.shape)

Shape after removing duplicates: (8790, 10)


In [12]:
text_columns = df.select_dtypes(include="str").columns

print("Text columns:")
print(text_columns.tolist())

for col in text_columns:
    df[col] = df[col].str.strip()

Text columns:
['show_id', 'type', 'title', 'director', 'country', 'rating', 'duration', 'listed_in']


In [13]:
df["date_added"] = pd.to_datetime(
    df["date_added"],
    errors="coerce"
)

print(df["date_added"].dtype)

datetime64[us]


In [14]:
df["added_year"] = df["date_added"].dt.year
df["added_month"] = df["date_added"].dt.month
df["added_day"] = df["date_added"].dt.day

df[["date_added", "added_year", "added_month", "added_day"]].head()

,date_added,added_year,added_month,added_day
0,2021-09-25,2021,9,25
1,2021-09-24,2021,9,24
2,2021-09-24,2021,9,24
3,2021-09-22,2021,9,22
4,2021-09-24,2021,9,24


In [16]:
df["duration_value"] = pd.to_numeric(
    df["duration"].str.extract(r"(\d+)")[0],
    errors="coerce"
)

df["duration_type"] = df["duration"].str.extract(
    r"(min|Seasons?)",
    expand=False
)

In [17]:
df[[
    "duration", "duration_value", "duration_type"
]].head(10)

,duration,duration_value,duration_type
0,90 min,90,min
1,1 Season,1,Season
2,1 Season,1,Season
3,91 min,91,min
4,125 min,125,min
5,9 Seasons,9,Seasons
6,104 min,104,min
7,87 min,87,min
8,127 min,127,min
9,76 min,76,min


In [18]:
print("Final Shape:", df.shape)

print("\nMissing Values:")
print(df.isnull().sum())

print("\nDuplicate Rows:")
print(df.duplicated().sum())

Final Shape: (8790, 15)

Missing Values:
show_id           0
type              0
title             0
director          0
country           0
date_added        0
release_year      0
rating            0
duration          0
listed_in         0
added_year        0
added_month       0
added_day         0
duration_value    0
duration_type     0
dtype: int64

Duplicate Rows:
0


In [19]:
df.to_csv("cleaned_dataset.csv", index=False)

print("Cleaned dataset saved successfully!")

Cleaned dataset saved successfully!


In [20]:
df["duration"].value_counts().head(20)

duration
1 Season     1791
2 Seasons     421
3 Seasons     198
90 min        152
97 min        146
94 min        146
93 min        146
91 min        144
95 min        137
96 min        130
92 min        129
102 min       122
98 min        120
99 min        118
88 min        116
101 min       116
103 min       114
106 min       111
100 min       108
89 min        106
Name: count, dtype: int64

In [21]:
df["duration_value"] = pd.to_numeric(
    df["duration"].str.extract(r"(\d+)")[0],
    errors="coerce"
)

df["duration_type"] = df["duration"].str.extract(
    r"(min|Seasons?)",
    expand=False
)

df[["duration", "duration_value", "duration_type"]].head(20)

,duration,duration_value,duration_type
0,90 min,90,min
1,1 Season,1,Season
2,1 Season,1,Season
3,91 min,91,min
4,125 min,125,min
5,9 Seasons,9,Seasons
6,104 min,104,min
7,87 min,87,min
8,127 min,127,min
9,76 min,76,min


In [22]:
df["duration_type"].value_counts()

duration_type
min        6126
Season     1791
Seasons     873
Name: count, dtype: int64

In [23]:
df["duration_type"] = df["duration_type"].replace(
    {"Season": "Seasons"}
)

df["duration_type"].value_counts()

duration_type
min        6126
Seasons    2664
Name: count, dtype: int64

In [24]:
print("Missing duration values:", df["duration_value"].isna().sum())
print("Missing duration types:", df["duration_type"].isna().sum())

Missing duration values: 0
Missing duration types: 0


In [25]:
df["country"].value_counts().head(20)

country
United States     3240
India             1057
United Kingdom     638
Pakistan           421
Not Given          287
Canada             271
Japan              259
South Korea        214
France             213
Spain              182
Mexico             138
Egypt              123
Australia          114
Turkey             112
Nigeria            105
Germany            104
China              100
Brazil              88
Taiwan              86
Indonesia           86
Name: count, dtype: int64

In [26]:
df["country"].value_counts().tail(10)

country
Iran           1
Slovenia       1
Guatemala      1
Jamaica        1
Somalia        1
Luxembourg     1
Senegal        1
Belarus        1
Puerto Rico    1
Cyprus         1
Name: count, dtype: int64

In [27]:
print("Not Given:", (df["country"] == "Not Given").sum())

Not Given: 287


In [28]:
df["listed_in"].value_counts().head(20)

listed_in
Dramas, International Movies                         362
Documentaries                                        359
Stand-Up Comedy                                      334
Comedies, Dramas, International Movies               274
Dramas, Independent Movies, International Movies     252
Kids' TV                                             219
Children & Family Movies                             215
Children & Family Movies, Comedies                   201
Documentaries, International Movies                  186
Dramas, International Movies, Romantic Movies        180
Comedies, International Movies                       176
Comedies, International Movies, Romantic Movies      152
Dramas                                               137
Dramas, International Movies, Thrillers              134
Action & Adventure, Dramas, International Movies     132
Action & Adventure                                   128
International TV Shows, TV Dramas                    121
Comedies, Dramas, Ind

In [29]:
df["primary_genre"] = df["listed_in"].str.split(",").str[0].str.strip()

df[["listed_in", "primary_genre"]].head(20)

,listed_in,primary_genre
0,Documentaries,Documentaries
1,"Crime TV Shows, International TV Shows, TV Action & Adventure",Crime TV Shows
2,"TV Dramas, TV Horror, TV Mysteries",TV Dramas
3,"Children & Family Movies, Comedies",Children & Family Movies
4,"Dramas, Independent Movies, International Movies",Dramas
5,"British TV Shows, Reality TV",British TV Shows
6,"Comedies, Dramas",Comedies
7,"Children & Family Movies, Comedies, Music & Musicals",Children & Family Movies
8,"Dramas, International Movies",Dramas
9,"Children & Family Movies, Music & Musicals",Children & Family Movies


In [30]:
df["primary_genre"].value_counts().head(20)

primary_genre
Dramas                      1599
Comedies                    1210
Action & Adventure           859
Documentaries                829
International TV Shows       773
Children & Family Movies     605
Crime TV Shows               399
Kids' TV                     385
Stand-Up Comedy              334
Horror Movies                275
British TV Shows             252
Docuseries                   220
Anime Series                 174
International Movies         128
Reality TV                   120
TV Comedies                  119
Classic Movies                80
TV Dramas                     67
Thrillers                     65
Movies                        53
Name: count, dtype: int64

In [31]:
print("Final Dataset Shape:", df.shape)

print("\nMissing Values:")
print(df.isnull().sum())

print("\nDuplicate Rows:", df.duplicated().sum())

Final Dataset Shape: (8790, 16)

Missing Values:
show_id           0
type              0
title             0
director          0
country           0
date_added        0
release_year      0
rating            0
duration          0
listed_in         0
added_year        0
added_month       0
added_day         0
duration_value    0
duration_type     0
primary_genre     0
dtype: int64

Duplicate Rows: 0


In [32]:
df.dtypes

show_id                      str
type                         str
title                        str
director                     str
country                      str
date_added        datetime64[us]
release_year               int64
rating                       str
duration                     str
listed_in                    str
added_year                 int32
added_month                int32
added_day                  int32
duration_value             int64
duration_type                str
primary_genre             object
dtype: object

In [33]:
df.to_csv("cleaned_dataset.csv", index=False)

print("Cleaned dataset saved successfully!")
print("Final shape:", df.shape)

Cleaned dataset saved successfully!
Final shape: (8790, 16)


In [34]:
import os

print("File exists:", os.path.exists("cleaned_dataset.csv"))

File exists: True


# Conclusion

The dataset was successfully inspected, cleaned, and preprocessed for further data analysis.

### Final Dataset Summary

- Original dataset size: 8,790 rows × 10 columns
- Final dataset size: 8,790 rows × 16 columns
- Missing values: 0
- Duplicate records: 0
- Converted `date_added` into datetime format
- Extracted year, month, and day from `date_added`
- Extracted numeric duration values from the `duration` column
- Standardized `Season` and `Seasons` into a single category
- Created a `primary_genre` feature from the `listed_in` column
- Preserved the original columns while adding useful analytical features

The cleaned dataset has been saved as `cleaned_dataset.csv` and is ready for further analysis and machine learning tasks.